File: 11-extra-example/SpringStudentBackend/src/main/java/com/makarands/service/StudentService.java
package com.makarands.service;

import java.util.List;

import com.makarands.model.Department;
import com.makarands.model.Student;

public interface StudentService {
	public boolean addStudent(Student student);
	public List<Student> findAllStudents();
	public boolean modifyStudent(Student student);
	public boolean removeStudent(int studentId);
	public Student findStudentById(int studentId);
	public List<Department> findAllDepartments();
}

File: 13-react-session-storage/src/service/StudentService.js
import axios from "axios";

export class StudentService {
    baseUrl = `http://localhost:9091/student-api/students`;

    checkLogin(login) {
        return axios.post(this.baseUrl + '/login', login)
    }
    addStudent(student) {
        return axios.post(this.baseUrl + '/', student);
    }
    findStudentById(studentId) {
        return axios.get(`${this.baseUrl}/${studentId}`);
    }
    findAllStudents() {
        return axios.get(this.baseUrl + '/');
    }
    deleteStudentById(studentId) {
        return axios.delete(`${this.baseUrl}/${studentId}`);
    }
    modifyStudent(student) {
        return axios.put(this.baseUrl + '/', student);
    }

    getAllDepartment() {
        return axios.get(`${this.baseUrl}/departments/`);
    }
}